mlflow/utils/pydantic_utils.py

```diff
@@ -7,6 +7,17 @@
 IS_PYDANTIC_V2_OR_NEWER = Version(pydantic.VERSION).major >= 2
 
 
+def validator(field:str, pre: bool=True):
```

Comment:
I would recommend the validator to follow the "new kwargs from v2, using `mode=`" and adapt it for V1 (and not the other way around), so that whenever V1 gets removed it is just removing that adaptation rather than modifying every validator.

right now it seems a working patch, but it could become a easier journey for the future